sound/soc/sof/intel/cnl.c

```diff
@@ -72,6 +72,9 @@ irqreturn_t cnl_ipc4_irq_thread(int irq, void *context)
 				snd_sof_ipc_get_reply(sdev);
 				snd_sof_ipc_reply(sdev, data->primary);
 
+				/* Let DSP know that we have finished processing the message */
```

Comment:
So, the argument is that we should use the ipc_lock spinlock to ack the received reply? What about the notification, should we use the spinlock there as well?